tapestry.js

```diff
@@ -1074,6 +1146,11 @@ function setDatasetProgress(progressObj) {
     
     }
     
+    // Update progress if fetched from database
+    if (tapestryWpUserId) {
```

Comment:
Why would you not update the progress if fetched from a cookie?

Reply:
Fetching the cookie seems to be synchronous, so d3 will update the arc for the progress. But with fetching from the database, it's asynchronous, so I'll need to explicitly call the updateViewedProgress function again so that it renders the progress.